### DISCUSSION TITLE
Adding a regex to email validation rule to prevent invalid emails registered

### DISCUSSION THREAD

**Discussion by assghard** [laravel/framework#36210] [Ideas] [answered]
Hello.
What do you think about adding this regex to email validation rule
`'regex:/^([a-z0-9\+_\-]+)(\.[a-z0-9\+_\-]+)*@([a-z0-9\-]+\.)+[a-z]{2,8}$/ix'`

Currently user can register typing an email like `john@gmail` and it will be passed throught `'email'` rule. But it is not a valid email. 
This regex in email verification describes a valid email address. It should contain a dot and from 2 to 8 letters after dot. 

The validation of register request is in **/app/Http/Controllers/Auth/RegisterController.php** so every new project I add this regex to the `Validator`. 

Do you know a better solution? 
Please consider adding this regex to **'email'** validation rule

**Comment by nsetyo** [comment, marked as answer]
@assghard You could use `email:rfc,filter` as rule

**Reply by pultho** [reply]
@nsetyo how do you translate the validation messages for this rules in your validation.php? 
See my question in: https://github.com/laravel/framework/discussions/36282

**Reply by ravindrajain-hash** [reply]
use this one working
'email'             =>  'required|string|regex:/^([a-z0-9\+_\-]+)(\.[a-z0-9\+_\-]+)*@([a-z0-9\-]+\.)+[a-z]{2,6}$/ix',
